base/loading.jl

```diff
@@ -408,9 +408,28 @@ end
 ## generic project & manifest API ##
 
 const project_names = ("JuliaProject.toml", "Project.toml")
-const manifest_names = ("JuliaManifest.toml", "Manifest.toml")
+const default_manifest_names = ("JuliaManifest.toml", "Manifest.toml")
 const preferences_names = ("JuliaLocalPreferences.toml", "LocalPreferences.toml")
 
+function manifest_names()
+    env_var_name = "JULIA_MANIFEST_NAME"
+    custom_def = strip(get(ENV, env_var_name, ""))
+    if !isempty(custom_def)
+        if !endswith(custom_def, ".toml")
+            throw(ArgumentError("Manifest name provided by `$env_var_name` must end with `.toml`. Provided: $(repr(custom_def))"))
+        end
+        man_name = expand_version_placeholders(custom_def)
+        if occursin("#", man_name)
+            throw(ArgumentError("""
+                Auto version population of `JULIA_MANIFEST_NAME` failed. Some `#` characters were not replaced: $(repr(name))
+                A maximum of three # chars may be used."""))
+        end
+        return (man_name,)
```

Comment:
I think it would be nice to make the `manifest_names()` function type-stable, which we can accomplish by making sure that the return value is always of the type `Tuple{String, String}`.

```suggestion
        return (man_name, man_name)
```

Reply:
It was mentioned in triage that these should actually just be vectors, so I just changed them over

Reply:
Hmmm. Do you remember the justification for why they should be vectors? It seems to me that they should be immutable, and therefore tulles.

Reply:
No. Them being immutable is a good point. I'll revert and do this